### PR TITLE
check whether voice file was successfully set to the media player

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Fixes https://github.com/mapbox/mapbox-navigation-android/issues/3750 by passing the voice instructions file reference via a file descriptor instead of a raw path. I was able to multiple successful runs in a row after this change which indicates that the issue is resolved. However, the `MediaPlayer` was never able to actually tell me what was causing the original issue. The docs mention, that if a raw path is provided the player will try to open the file on a separate thread which probably isn't the case with the file descriptor. We'll need to monitor this. The change also refactors error handling in case the file actually fails to be loaded to completely prevent the crash. Now, if the file is corrupt or fails to load, it just won't be played.

### Changelog

```
<changelog>Fixes an issue where a failed voice instructions file loading would crash the library.</changelog>
```

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java [hunk 1]
```diff
@@ -160,17 +162,28 @@ private void onError(String errorText) {
...
   private void playInstruction(@NonNull File instruction) {
-    setupMediaPlayer(instruction.getPath());
+    setupMediaPlayer(instruction);
   }
 
-  private void setupMediaPlayer(String instructionPath) {
-    if (TextUtils.isEmpty(instructionPath)) {
+  private void setupMediaPlayer(@NonNull File instruction) {
+    if (!instruction.canRead()) {
+      String msg = "Cannot read file " + instruction.getAbsolutePath();
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java:170]: Extract this magic number into a constant?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java [hunk 2]
```diff
@@ -160,17 +162,28 @@ private void onError(String errorText) {
...
-    mediaPlayer.prepareAsync();
-    addListeners();
+
+    try (FileInputStream fis = new FileInputStream(instruction)) {
+      mediaPlayer = new MediaPlayer();
+      setDataSource(fis);
+      mediaPlayer.prepareAsync();
+      addListeners();
+    } catch (FileNotFoundException ex) {
+      String msg = "Instructions file not found for " + instruction.getAbsolutePath() + ". " + ex.getMessage();
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java:181]: Same as above re: magic number

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java [hunk 3]
```diff
@@ -160,17 +162,28 @@ private void onError(String errorText) {
...
+      Timber.e(ERROR_TEXT, msg);
+    } catch (IOException ex) {
+      String msg = "Error for " + instruction.getAbsolutePath() + ". " + ex.getMessage();
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java:184]: Same as above re: magic number

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java [hunk 4]
```diff
@@ -180,15 +193,22 @@ private void pauseInstruction() {
...
     } catch (IOException ioException) {
       Timber.e(ERROR_TEXT, ioException.getMessage());
     }
   }
 
   private void addListeners() {
+    mediaPlayer.setOnErrorListener(new MediaPlayer.OnErrorListener() {
+      @Override
+      public boolean onError(MediaPlayer mp, int what, int extra) {
+        Timber.e(ERROR_TEXT, "code_what: " + what + " code_extra: " + extra);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java:208]: Same as above re: magic number

**Review by Guardiola31337 — APPROVED** [review]

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java (hunk 1 above)
Comment (LukasPaczos) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java:170]: Not really a magic number, is it? String speaks for iteslf.

Comment (Guardiola31337) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/MapboxSpeechPlayer.java:170]: Debatable 👍

**Pull request merged**